external/fv3fit/fv3fit/reservoir/train.py

```diff
@@ -275,25 +293,65 @@ def train_reservoir_model(
             transformers=transformers,
             hybrid_input_mask=hybrid_input_mask_array,
         )
-        adapter = HybridReservoirDatasetAdapter(  # type: ignore
+        adapter_model = HybridReservoirDatasetAdapter(
             model=model,
             input_variables=model.input_variables,
             output_variables=model.output_variables,
         )
 
-    if validation_batches is not None and wandb.run is not None:
-        try:
-            ds_val = validation_prediction(
-                model,
-                val_batches=validation_batches,
-                n_synchronize=hyperparameters.n_timesteps_synchronize,
+    if wandb.run is not None and validation_batches is not None:
+        if not hyperparameters.validate_sst_only:
+            try:
+                ds_val = validation_prediction(
+                    model,
+                    val_batches=validation_batches,
+                    n_synchronize=hyperparameters.n_timesteps_synchronize,
+                )
+                log_rmse_z_plots(ds_val, model.output_variables)
+                log_rmse_scalar_metrics(ds_val, model.output_variables)
+            except Exception as e:
+                logging.error("Error logging validation metrics to wandb", exc_info=e)
+        else:
+            data = next(iter(validation_batches))
+            input_data = process_validation_batch_data_to_dataset(
+                data, adapter_model.nonhybrid_input_variables
+            )
+
+            if adapter_model.is_hybrid:
+                adapter_model = cast(HybridReservoirDatasetAdapter, adapter_model)
+                hybrid_data = process_validation_batch_data_to_dataset(
+                    data, adapter_model.hybrid_variables, trim_divider=rank_divider
+                )
+            else:
+                hybrid_data = None
+
+            output_vars = list(adapter_model.output_variables)
+            if "mask_field" in data:
```

Comment:
Good catch.  I was using this because setting the hyperparameters configuration forced masking of reservoir inputs.  I switched to applying the mask in the transforms so I could be more particular about which fields are masked, even if they are all included in the same encoding.  I still wanted the validation output masked and this was a mechanism for that.  

I switched the implementation up so that we have to specify boolean flags to mask either/both of the reservoir and hybrid inputs.  Setting the `mask_variable` will now apply to the validation when specified.